src/test/java/pageObjects/OpportunityPage.java

```
@@ -20,4 +18,52 @@ public void clickRelatedTab() {
         waitUntilLocatorDisplayAndClick(relatedTab, 1, WAIT_TIME_IN_SECONDS);
         explicitWaitInSeconds(WAIT_TIME_IN_SECONDS).until(ExpectedConditions.numberOfElementsToBeMoreThan(content_section,15));
     }
+
+    public void openActionDropDown(String value) {
+
+        System.out.println(" == INFO: wait for openActionDropDown.............. ");  //Todo: remove before merge
+
+        waitUntilNumberOfElementsToBeMoreThan(actionsDropDown, 0);
+        waitUntilLocatorDisplayAndClick(actionsDropDown, 1, WAIT_TIME_IN_SECONDS);
+        click(By.cssSelector("[title=\"" + value + "\"]"));
+
+        System.out.println(" == INFO: click on: " + value);  //Todo: remove before merge
+    }
+
+    public class BillingAccountSetModal {
+
+        private By addNewBillingAccountBtn = By.cssSelector("button[title=\"Add New Billing Account\"]");
+        private By addBillingAccountModal = By.id("EnziModalCloseButton");
+        private By copyOpportunityMainContactLink = By.id("OppMainContact");
+        private By primaryMemberEmail = By.id("searchInput");
+        private By accoutName = By.id("AccountName");
+        private By saveButton = By.cssSelector("button[title=\"Save\"]");
+
+
+        public void accountSelector(String email) {
+
+            System.out.println(" == INFO: accountSelector section start:");  //Todo: remove before merge
+
+            try {
+                waitForElementToBeClickable(addNewBillingAccountBtn);
+                click(addNewBillingAccountBtn);
+                waitForElementToBeClickable(copyOpportunityMainContactLink);
+                if (findElement(copyOpportunityMainContactLink).isDisplayed()) {
```

i am not sure, but maybe an else statement / handling is missing here ?

---

the idea with tests is that it should fail in case mandatory element is missing
we don't want to make this test pass or wait for too long when element is missing

---

i added the requested else - thanks Ben!